File: scala/src/com/lut/scala/Day4/ExceptionTest.scala
package com.lut.scala.Day4

import scala.io.Source
import scala.util.{Failure, Success, Try}

/**
  * @author qsg
  * @Date 2019/11/29 9:55
  * @describe
  * @version 1.0
  **/
class ExceptionTest {
  def show(s:String):Unit = {
    try {
      if (s == "") throw new NullPointerException("空指针异常")
    }catch {
      case e1:NullPointerException =>e1.printStackTrace()
    }finally {
    println(s)
    }
  }
  @throws(classOf[Exception])
  def readFromFileOption(file:String):Option[List[String]]={
    try{
      Some(Source.fromFile(file).getLines().toList)
    }catch {
      case e:Exception => None
    }
  }
  def readFile(file:String):Try[List[String]]={
    Try(Source.fromFile(file).getLines().toList)
  }


}
object ExceptionTest{
  def main(args: Array[String]): Unit = {
//    new ExceptionTest().show("123")
//    var s = new ExceptionTest().readFromFileOption("src/testfile")
//    println(s)
    var s = new ExceptionTest().readFile("src/testfile")
    if(s.isSuccess){
      println("成功")
    }
    s match {
      case n:Success[List[_]]=>n.foreach(println(_))
      case m:Failure[List[_]]=>println("文件未找到")
    }
  }

}

File: homework/src/lut/day14/ExceptionTest.java
package lut.day14;

public class ExceptionTest {
    public static void main(String[] args) throws indexException {
        try {
            new A().show();
        }catch (indexException d){
            System.out.println("数组越界异常");
        }
    }
}
class A{
    public void show () throws indexException {

        try{
            int [] a = new int[10];
            System.out.println(a[10]);

        }catch (IndexOutOfBoundsException e){
            System.out.println(e);
           throw  new indexException();
        }

    }

}
class indexException extends Exception{
    indexException(){
      //new  IndexOutOfBoundsException().printStackTrace();
        System.out.println("数组越界异常");
    }
}
